src/Features/ERDDAP/utils/waterLevelThresholds.tsx

```diff
@@ -0,0 +1,51 @@
+import { colors } from "Shared/colors"
+
+export const getWaterLevelThresholdsMapRawComp = (floodThresholds) => {
+  if (floodThresholds) {
+    return floodThresholds.map((threshold) => {
+      const max_value =
+        threshold.name == "Major" ? null : floodThresholds.find((f) => f.name === compChart[threshold.name])?.min_value
+      return {
+        name: threshold.name,
+        min_value: threshold.min_value,
+        max_value: max_value,
+        description: threshold.description,
+      }
+    })
+  }
+  return null
+}
+
+const compChart = {
+  Action: "Minor",
+  Minor: "Moderate",
+  Moderate: "Major",
+}
+
+export const getSurpassedThreshold = (value, thresholds) => {
+  const range = thresholds.find((t) => {
+    if (t.name === "Major") {
+      return value >= t.min_value
+    }
+    return value >= t.min_value && value <= t.max_value
+  })
+  if (range) {
+    return range.name
+  }
+  return null
+}
+
```

Comment:
Theoretically, a way this can be calculated differently is by using an iterative recursive function. Iterating through the flood levels, and returning the flood level where the value is above the min_value but below the next thresholds min_value. However, this would assume the flood thresholds are in order within the array. This current way seems verbose, but also protects against out-of-order flood thresholds.